### DISCUSSION TITLE
[Question][Hw5] About node

### DISCUSSION THREAD

**Discussion by nubi12336** [prosyslab-classroom/cs348-information-security#628] [Q&A]
I thought when we run the code, we have to use `semantics.transfer_node`
for each node in given table (All labels have bottom memories) to get a new table.
Thus, I tried to do above thing but I failed.

In example1.c, there is `int x = source()`. In my thought, this line is a `node`, so it is in a given table. Let this `node` is `c`.
Then, in `transfer_node llctx c memory`, match `node` with `Atomic instr` and `instr` is  `int x = source()`.
In this `instr`, there is a first operand `x` and second operand `source()`. Therefore, I used `utils.is_source instr` and I added this new `x -> T` to memory(It was bottom memory).

Are there something wrong with above statements..?
When I tried to print out new memory, it occurs error `Cannot find lhs of declare dso_local i32 @source(...) #2
Fatal error: exception Not_found`.
Did I misunderstand about Node or Table?

**Comment by nubi12336** [comment]
I think just assignment is not working. `int x = 0` also occurs error.

**Comment by RiceBiscuits** [comment]
`x` is not a first operand. 

Please see this document. (https://github.com/prosyslab-classroom/llvm-primer/blob/master/data-structure.md)

**Comment by nubi12336** [comment]
I see. x is also just instr. Right?

**Comment by RiceBiscuits** [comment]
Yes.
